custom_scripts/py_scripts/db_scanner/db_scanner.py

```diff
@@ -0,0 +1,558 @@
+            OperationFailure,
+        ) as err:
+            print(f"Connection was over by server: {err}")
+            return err
+        return docs_num
+
+    def find_in_collection_with_reg(
+        self, client: database, collection_name: str
+    ) -> (dict or None, Exception or None):
+        """
+        Find values in collection using regular expressions for searching
+        :param client: MongoClient.database
+        :param collection_name: name of a collection in database
+        :return: founded result
+        """
+        resulted_value = self.get_number_of_documents(client, collection_name)
+        if not isinstance(resulted_value, int):
+            return None, resulted_value
+        docs_num = resulted_value
+        result_list = []
+        if not docs_num:
+            return None, None
+        keys = client[collection_name].find_one().keys()
+        keys_to_take = []
+        regulars = []
+        projection = {}
+        for key in keys:
+            if findall(DBScannerDefaultValues.NOT_INCLUDED, key):
+                projection.update({key: 0})
+                continue
+            for info in DBScannerDefaultValues.CRITICAL_INFORMATION:
+                if findall(info, key):
+                    keys_to_take.append(key)
+                    break
+            if key in keys_to_take:
+                continue
+            else:
+                regs = {
+                    key: {
+                        "$regex": DBScannerDefaultValues.REGULARS_VALUES
+                        + DBScannerDefaultValues.REGULARS_INFO,
+                        "$options": "i",
+                    }
+                }
+                regulars.append(regs)
+        if keys_to_take:
+            key_projection = {}
+            for key in keys_to_take:
+                key_projection.update({key: 1})
+            key_projection.update({"_id": 0})
+            result = client[collection_name].find(
+                projection=key_projection,
+                limit=100,
+            )
+            for doc in result:
+                converted = convert_dict_values_into_needed_type(doc)
+                if converted:
+                    if converted not in result_list:
+                        result_list.append(converted)
+            try:
+                self.check_time_of_connection()
+            except DBScannerConnectionTimeout as err:
+                if err:
+                    return result_list, err
+        result = [
+            client[collection_name].find(reg, projection, limit=100) for reg in regulars
+        ]
+        for cursor in result:
+            for doc in cursor:
+                res = find_values_by_key(doc)
+                if res:
+                    if res not in result_list:
+                        result_list.append(res)
+            try:
+                self.check_time_of_connection()
+            except DBScannerConnectionTimeout as err:
+                if err:
```

Comment:
Do we really need to check that? 😺 
I believe that we almost always have the `err` object. Or something special here?